### ISSUE TITLE
500 ISE when docker pull with missing project in the image name

### ISSUE THREAD

**Issue by wu105** [goharbor/harbor#12010]
If you are reporting a problem, please make sure the following information are provided:

**Expected behavior and actual behavior:**
suppose the correct image spec is `<harborsite>/<proj>/<image>:<tag>`
docker pull `<harborsite>/<proj>/<image>:<tag>`
works, and 
docker pull `<harborsite>/<image>:<tag>`
is expected to return errors about the image or parts of it not found

**Steps to reproduce the problem:**
Currently we are getting the following error:
`Get https://<harborsite>/<image>:<tag>: received unexpected HTTP status: 500 Internal Server Error`

**Versions:**
Please specify the versions of following systems.

- harbor version: [1.9.3x]
- docker engine version: [19.3.1]
- docker-compose version: [n/a]

**Additional context:**

- **Harbor config files:** You can get them by packaging `harbor.yml` and files in the same directory, including subdirectory.
- **Log files:** You can get them by package the `/var/log/harbor/` .

**Comment by bitsf** [comment]
We should report "project not found" but not 500
![image](https://user-images.githubusercontent.com/188115/82896812-277c1280-9f89-11ea-93ff-81ab60f89cac.png)

**Comment by bitsf** [comment]
We should report "project not found" but not 500
![image](https://user-images.githubusercontent.com/188115/82896812-277c1280-9f89-11ea-93ff-81ab60f89cac.png)

**Comment by reasonerjt** [comment]
see comment 
https://github.com/goharbor/harbor/issues/12614#issuecomment-674702400

After this change the `docker pull` command will see error like:
```
docker pull jt-dev.local.goharbor.io/notexist
Using default tag: latest
Error response from daemon: unknown: bad request: invalid repository name: notexist
```

This is acceptable b/c it is not a valid repo name to Harbor (as for now Harbor requires namespace/project name )

**Comment by reasonerjt** [comment]
Fixed in #12790
